rpcs3/Emu/RSX/Common/BufferUtils.cpp

```diff
@@ -755,39 +762,290 @@ namespace
 
 	struct primitive_restart_impl
 	{
-		template<typename T>
+		AVX2_FUNC
+		static
+		std::tuple<u16, u16, u32> upload_u16_swapped_avx2(const void *src, void *dst, u32 count, u16 restart_index)
+		{
+			u32 dst_index = 0;
+
+			const __m256i mask = _mm256_set_epi8(
```

Comment:
This mask looks suspicious because 256-bit PSHUFB instruction does two 128-bit PSHUFB in each lane, with same 4-bit indexing.

Reply:
I’ve dropped the AVX2 path from this PR, I’ve also found out that the count wasn’t right and if I change it I end up with big planes flying around, I’ll make a new PR later once I’ve figured out why that is, and to not keep this PR from being merged.

Reply:
In the end it indeed was that I didn’t understand lanes in AVX2!

The issue wasn’t here but in how I tried to use a mask to extract the higher lane into the lower lane, instead of just extracting it.

This function is now correctly processing 16 words at once, but it isn’t much faster than the SSE4.1 version. :(